From 6182a760fc9a770ec4fea438ac563ebbd3103666 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 17 Feb 2021 15:42:27 +0800
Subject: [PATCH] drm/rockchip: vop2: No setup alpha if fb is NULL

We find sometimes hwc will post a plane with NULL
fb, this is no need to setup alpha.

Change-Id: I1ca475e7171c08a4182a3a98bf357d3bf00d8fdc
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 3ed6531990b1..9fefa3d159b1 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -4128,6 +4128,8 @@ static void vop2_setup_cluster_alpha(struct vop2 *vop2, struct vop2_cluster *clu
 
 	if (top_win_vpstate) {
 		fb = top_win_vpstate->base.fb;
+		if (!fb)
+			return;
 		if (top_win_vpstate->base.pixel_blend_mode == DRM_MODE_BLEND_PREMULTI)
 			premulti_en = true;
 		else
@@ -4135,6 +4137,8 @@ static void vop2_setup_cluster_alpha(struct vop2 *vop2, struct vop2_cluster *clu
 		src_pixel_alpha_en = is_alpha_support(fb->format->format);
 	}
 	fb = bottom_win_vpstate->base.fb;
+	if (!fb)
+		return;
 	dst_pixel_alpha_en = is_alpha_support(fb->format->format);
 	alpha_config.src_premulti_en = premulti_en;
 	alpha_config.dst_premulti_en = false;
-- 
2.35.3

